src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -919,8 +924,8 @@ protected DynamoModel(IStartConfiguration config)
 
             CustomNodeManager = new CustomNodeManager(NodeFactory, MigrationManager, LibraryServices);
 
-            LuceneSearchUtility = new LuceneSearchUtility(this);
-            LuceneSearchUtility.InitializeLuceneConfig(LuceneConfig.NodesIndexingDirectory);
+            LuceneSearch.LuceneSearchUtility = new LuceneSearchUtility(this);
```

Comment:
Also for this case I was thiking that the LuceneSearchUtility creation (new LuceneSearchUtility(this)) and Initialization  should be done inside the singleton because otherwise we can create many instances of LuceneSearchUtility, in the case of Node Autocomplete we create it and dispose the writer everytime the user is trying to search (due that we are indexing a different small set of nodes everytime is triggered), then probably is not useful in that case (or we will need to modify the code to create the LuceneSearchUtility for node autocomplete once and at the end delete the indexed info so we won't be creating the instance several times).
Not sure if this is a good idea is just a set thoughts that crossed my mind when thinking about implementing a Singleton for Lucene creation. Thanks

Reply:
Thanks Roberto, working on these changes.

Reply:
@QilongTang @reddyashish 
I was checking this case and I think that is not possible due that in the LuceneSearchUtility constructor we need to pass the DynamoModel and for the InitializeLuceneConfig method we are passing different parameters depending if is for PackageManager, Nodes or NodeAutocomplete, so is better leave it in this way.
Thanks

Reply:
@RobertGlobant20 Thanks for confirming